tests/telemetry/conftest.py

```diff
@@ -50,23 +84,28 @@ def setup_streaming_telemetry(duthosts, enum_rand_one_per_hwsku_hostname, localh
     """
     try:
         duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+        has_gnmi_config = check_gnmi_config(duthost)
+        if not has_gnmi_config:
+            create_gnmi_config(duthost)
+        env = GNMIEnvironment(duthost)
         default_client_auth = setup_telemetry_forpyclient(duthost)
 
         if default_client_auth == "true":
-            duthost.shell('sonic-db-cli CONFIG_DB HSET "TELEMETRY|gnmi" "client_auth" "false"',
+            duthost.shell('sonic-db-cli CONFIG_DB HSET "%s|gnmi" "client_auth" "false"' % (env.gnmi_config_table),
                           module_ignore_errors=False)
-            duthost.shell("systemctl reset-failed telemetry")
-            duthost.service(name="telemetry", state="restarted")
+            duthost.shell("systemctl reset-failed %s" % (env.gnmi_container))
+            duthost.service(name=env.gnmi_container, state="restarted")
         else:
             logger.info('client auth is false. No need to restart telemetry')
 
         # Wait until telemetry was restarted
-        py_assert(wait_until(100, 10, 0, duthost.is_service_fully_started, "telemetry"), "TELEMETRY not started.")
+        py_assert(wait_until(100, 10, 0, duthost.is_service_fully_started, env.gnmi_container),
```

Comment:
In err message, replace TELEMETRY with env.gnmi_container

Reply:
Updated.